splight_lib/models/hub.py

```diff
@@ -147,15 +147,36 @@ def delete(self):
 
     def download(self):
         hub_client = get_hub_client()
-        params = {"name": self.name, "version": self.version}
-        return hub_client.download(data=params)
+        return hub_client.download(id=self.id, name=self.name, type="source")
+
+    def save(self):
+        hub_client = get_hub_client()
+        return hub_client.save(instance=self)
+
+    def build(self):
+        hub_client = get_hub_client()
+        return hub_client.build(id=self.id)
 
     @classmethod
     def upload(cls, path: str):
         hub_client = get_hub_client()
         spec = get_spec(path)
         name = spec["name"]
         version = spec["version"]
+        raw_hub_component = hub_client.get(
+            name=name, version=version, first=True
+        )
+        if not raw_hub_component:
+            raw_hub_component = HubComponent(
+                name=name,
+                version=version,
+                splight_lib_version=spec.get("splight_lib_version", None),
+                splight_cli_version=spec.get("splight_cli_version", None),
+                privacy_policy=spec.get("privacy_policy"),
+                component_type=spec.get("component_type"),
+            ).save()
```

Comment:
What is happening with routines, input, and other confirations of the hub component, are those needed?